### ISSUE TITLE
Nuke small_array_adapter (and replace it)

### ISSUE THREAD

**Issue by Morwenn** [Morwenn/cpp-sort#16]
It saddens me bit since it is was the original motivation for **cpp-sort**, but `small_array_adapter` has to go: it's too specific and too too hard to maintain. For example, for an array of size 10, one sorting algorithm was faster for `int` and `long int`, another one for `float`, another one for `double` and yet another one for `long long int` and `long double`. And we're still only talking about a small  subset of the built-in types, and those results might change depending on the architecture. In other words, maintaining a sorter whose goal is *raw speed* is impossible. Therefore, `small_array_adapter` has to go. However, it should provide tools to let developers create an equivalent adapter if needed, here is what is needed:
* `sorting_network_adapter`: a small arrays adapter that implements sorting network algorithms with possibly the smallest number of conditional swaps. Sorting networks seem to be among the fastest algorithms when it comes to sorting small arrays of integers.
* an adapter tool that allows to provide different algorithms for different sizes of arrays.

Basically, we should provide to the users simple tools that will allow them to find the most suitable small-arrays sorting algorithm depending on their needs.
